core/src/main/java/org/stellar/anchor/asset/JsonAssetService.java

```diff
@@ -19,12 +19,22 @@ public List<AssetInfo> listAllAssets() {
     return new ArrayList<>(assets.assets);
   }
 
-  public AssetInfo getAsset(String code, String issuer) {
+  public AssetInfo getAsset(String code) {
     for (AssetInfo asset : assets.assets) {
       if (asset.getCode().equals(code)) {
-        if (issuer == null || issuer.equals(asset.getIssuer())) {
           return asset;
-        }
+      }
+    }
+    return null;
+  }
+
+  public AssetInfo getAsset(String code, String issuer) {
+    if (issuer == null) {
+      return getAsset(code);
+    }
```

Comment:
nit: the design I see more often in libraries and SDKs is for the method with fewer parameters to call the one with the most parameters, for instance:

```java
public AssetInfo getAsset(String code) {
  return getAsset(code, null);
}
```

Nothing wrong with your design though, we can move on with that.